cmd/kube-apiserver/app/server.go

```diff
@@ -346,12 +364,12 @@ func (s *APIServer) Run(_ []string) error {
 	if err != nil {
 		return err
 	}
-	storageVersions := make(map[string]string)
-	if s.StorageVersion == "" {
-		s.StorageVersion = g.Version
+
+	storageVersions := generateStorageVersionMap(s.StorageVersion, s.StorageVersions)
+	if _, found := storageVersions[g.Group]; !found {
+		storageVersions[g.Group] = g.Version
```

Comment:
How about we make a sanity check in generateStorageVersionMap when parsing s.StorageVersions?